tools/ReleaseNotes/release_editor.py

```diff
@@ -0,0 +1,126 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import os
+import sys
+import shutil
+
+
+def getReleaseRoot():
+    script_dir = os.path.split(sys.argv[0])[0]
+    return os.path.abspath(os.path.join(script_dir, '../../docs/source/release/'))
+
+
+def fixReleaseName(name):
+    if not name.startswith('v'):
+        name = 'v' + name
+    return name
+
+
+def createFileLocation(releaseNo):
+    release_root = getReleaseRoot()
+    release_root = release_root + '/' + releaseNo
+    return release_root
+
+
+def updateFile(originalFile, replaceText, textToReplace):
+    # opens the upper level release note e.g. diffraction.rst
+    mainRST = open(originalFile, 'r')
+    rstToRead = mainRST.read()
+    mainRST.close()
+
+    # replaces the amalgamate directive with the notes compiled for that heading
+    newdata = rstToRead.replace(textToReplace, replaceText)
+
+    mainRST = open(originalFile, 'w')
+    mainRST.write(newdata)
+    mainRST.close()
+
+
+def createFileName(fileName, pathDirectory, includeStatement):
+    newFileName = fileName.replace(includeStatement, "")
+    newFileName = newFileName.replace("\n", "")
+    newFileName = pathDirectory + '/' + newFileName
+    return os.path.normpath(newFileName)
+
+
+def addReleaseNotesToMain(pathDirectory, listOfDirectories):
+    for file in os.listdir(pathDirectory):
+        if file.endswith(".rst"):
+            with open(mainDirectory + '/' + file) as fileToEdit:
+                for line in fileToEdit:
+                    if line.startswith(directive):
+                        fileName = createFileName(line, pathDirectory, directive)
+                        if fileName in listOfDirectories:
+                            releaseNotes = collateNotes(fileName)
+                        else:
+                            releaseNotes = ""
+                        originalFile = mainDirectory + '/' + file
+                        updateFile(originalFile, releaseNotes, line)
```

Comment:
This seems a little too nested to be easily readable, could it be refactored?

Reply:
I wasn't sure about doing the if/else and the listOfDirectories. It could potentially save time but actually the whole thing doesn't take very long anyway so I could probably remove all that which would make it a little less nested.

Not sure if it can be refactored easily otherwise. I had great difficulty accessing the right part of the file to replace.